### PR TITLE
[Enhance] limit maximum length of JSON string to 16MB

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] bugfix
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Limit the maximum length of JSON string to 16MB, to avoid memory allocation and storage issues.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.2 branch-2.3 branch-2.4

**Review by ZiheLiu — APPROVED** [review]

File: be/src/util/json.h [hunk 1]
```diff
@@ -29,6 +29,9 @@ enum JsonType {
     JSON_OBJECT = 5,
 };
 
+// Maximum length of JSON string is 16MB
+constexpr size_t kJSONLengthLimit = 16 << 20;
```
Comment (kangkaisen) [inline be/src/util/json.h:33]: Do we need to make it to be configurable？

Comment (mofeiatwork) [reply be/src/util/json.h:33]: It's a defense for system safety, cause we could not support a huge JSON string

If someday we could support a huge JSON string, just remove this limitation.

Anyway, nobody want to configure this limitation.

**Comment by mofeiatwork** [comment]
run starrocks_be_unittest

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Pull request merged**